### ISSUE TITLE
Extensions for platform-specific features should not show up on other platforms

### ISSUE THREAD

**Issue by sanketj** [MicrosoftEdge/JsDbg#86]
For extensions that target platform-specific features, like WinD3D11AngleTextures or TSFTextStore, we should come up with a declarative way to tag them as platform-specific and "hide" them on other platforms. The "targetModules" field in the extension manifest achieves this for extension that are in platform-specific modules. But for situations where the extension targets a generic module, like msedge_child or chrome_child, this doesn't work. One option is to add a "targetPlatforms" field to the extension manifest.

**Comment by cbiesinger** [comment]
I wonder if you could achieve some of this with a "requireModule" attribute. For windows, specify ntdll, for linux, maybe "c" or "gtk". I'm sure mac has something.

**Comment by sanketj** [comment]
> I wonder if you could achieve some of this with a "requireModule" attribute. For windows, specify ntdll, for linux, maybe "c" or "gtk". I'm sure mac has something.

Yes, that is one way to detect which platform the user is currently debugging on.
